hawc/apps/common/dsstox.py

```diff
@@ -0,0 +1,36 @@
+import base64
+import logging
+
+import requests
+
+logger = logging.getLogger(__name__)
+
+
+def fetch_dsstox(cas_number):
+    d = {}
+    try:
+        # get details
+        url = r"https://actorws.epa.gov/actorws/chemIdentifier/v01/resolve.json"
+        params = {"identifier": cas_number}
+        response_dict = requests.get(url, params).json()["DataRow"]
+        d["CommonName"] = response_dict["preferredName"]
+        d["SMILES"] = response_dict["smiles"]
+        d["MW"] = response_dict["molWeight"]
+        d["DTXSID"] = response_dict["dtxsid"]
+
+        # get image
+        url = r"https://actorws.epa.gov/actorws/chemical/image"
```

Comment:
you might not need this request; the original request includes an image column with a url in it to the image - https://actorws.epa.gov/actorws/chemIdentifier/v01/resolve?identifier=80-05-7

```html
<tr>
	<td>Image</td>
	<td><img src="http://actorws.epa.gov/actorws/chemical/image?dtxcid=DTXCID30182&amp;w=150&amp;h=150&amp;fmt=svg" alt="Structure not found"></td>
</tr>
```

Might be better to use this url instead of another request, so we only hit the remote server 2 times instead of 3

Reply:
The DSSTox database will be hit twice regardless, once for details and a second time for image data. This current implementation allows us to cache the image data instead of the image url, which would save users from unnecessarily making repeat requests for the image.

Reply:
oh you're right, it requires 2 hits either way. Ok, keeping as is works!